### PR TITLE
Feature/124 update application page

### PR DESCRIPTION
**Pull request by RohitGarudadri**
# Description

Updated the application page by refactoring the hacker model for the front and backend by grouping together similar properties. Also changed certain fields (i.e. major -> fieldOfStudy) and added new question to essay section.

Fixes # (124)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- [x] NPM run tests all passed and manually tested

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by pierreTklein** [comment]
Congrats on your first PR! Seems like a your PR involves a lot of formatting changes. Please make sure that this isn't the case-- it's basically impossible to tell what logic changes are being made here otherwise! Thanks :).

File: middlewares/validators/validator.helper.js [hunk 1]
```diff
@@ -198,8 +204,9 @@ function stringValidator(fieldLocation, fieldname, optional = true) {
     const name = setProperValidationChainBuilder(
         fieldLocation,
         fieldname,
-        "invalid string"
+        "Invalid String"
```
Comment (MrHarshPatel) [inline middlewares/validators/validator.helper.js]: Nit: Maybe change others as well to capitalize, or leave them all uncapitalized?

File: middlewares/validators/hacker.validator.js [hunk 2]
```diff
@@ -6,15 +6,44 @@ module.exports = {
     newHackerValidator: [
         // status will be added automatically
         VALIDATOR.mongoIdValidator("body", "accountId", false),
-        VALIDATOR.stringValidator("body", "school", false),
-        VALIDATOR.stringValidator("body", "degree", false),
-        VALIDATOR.stringValidator("body", "gender", false),
-        VALIDATOR.booleanValidator("body", "needsBus", false),
+        VALIDATOR.stringValidator("body", "application.general.school", false),
```
Comment (MrHarshPatel) [inline middlewares/validators/hacker.validator.js]: Does application need to be validated as an object before this?

File: middlewares/validators/validator.helper.js [hunk 3]
```diff
@@ -438,37 +457,48 @@ function applicationValidator(fieldLocation, fieldname, optional = true) {
...
+                    !app.general.URL.github ||
+                    typeof app.general.URL.github === "string";
+                hasValid.dribbble =
+                    !app.general.URL.dribbble ||
+                    typeof app.general.URL.dribbble === "string";
                 hasValid.personal =
-                    !app.portfolioURL.personal ||
-                    typeof app.portfolioURL.personal === "string";
+                    !app.general.URL.personal ||
+                    typeof app.general.URL.personal === "string";
```
Comment (MrHarshPatel) [inline middlewares/validators/validator.helper.js]: should stringValidator be used for these fields instead?

File: tests/hacker.test.js [hunk 4]
```diff
@@ -562,11 +597,13 @@ describe("PATCH update one hacker", function() {
                 agent.close();
                 return done(error);
             }
+            let app = TeamHacker0.application;
+            app.other.gender = "Other";
```
Comment (MrHarshPatel) [inline tests/hacker.test.js]: Nit: Maybe send application: {other: {gender: "Other"}}} ?

Comment (pierreTklein) [reply tests/hacker.test.js]: I agree. We shouldn't mutate the test data.

File: package.json [hunk 5]
```diff
@@ -26,11 +26,14 @@
...
     "express-winston": "^2.6.0",
-    "handlebars": "^4.4.3",
+    "handlebars": "^4.5.3",
     "jsonwebtoken": "^8.5.1",
+    "lodash": "^4.17.15",
+    "lodash.clonedeep": "^4.5.0",
     "memory-cache": "^0.2.0",
     "mongoose": "^5.7.5",
     "multer": "^1.4.2",
+    "nvm": "0.0.4",
```
Comment (loreina) [inline package.json]: nvm should be installed on your system, not in the repo

File: package.json [hunk 6]
```diff
@@ -26,11 +26,14 @@
     "express": "~4.17.1",
     "express-validator": "^6.2.0",
     "express-winston": "^2.6.0",
-    "handlebars": "^4.4.3",
+    "handlebars": "^4.5.3",
     "jsonwebtoken": "^8.5.1",
+    "lodash": "^4.17.15",
```
Comment (loreina) [inline package.json]: why do we need lodash?

File: package.json [hunk 7]
```diff
@@ -26,11 +26,14 @@
...
+    "lodash.clonedeep": "^4.5.0",
```
Comment (loreina) [inline package.json]: ditto

File: tests/hacker.test.js [hunk 8]
```diff
@@ -8,6 +8,7 @@ const should = chai.should();
 const Hacker = require("../models/hacker.model");
 const fs = require("fs");
 const path = require("path");
+// const cloneDeep = require("lodash/clonedeep");
```
Comment (loreina) [inline tests/hacker.test.js]: Is this commented out for future use?

**Pull request merged**